src/helpers/strings.ts

```diff
@@ -74,10 +86,47 @@ export function formatNumber(
 type CurrencyFormatterOptions = {
   decimals?: number;
   valueIfNaN?: string;
-  currency?: NativeCurrencyKey;
+  currency: NativeCurrencyKey;
 };
 
-const toSubscript = (str: string | number) => str.toString().replace(/[0-9]/g, num => String.fromCharCode(0x2080 + +num));
+const toSubscript = (digit: string): string => {
+  'worklet';
+  if (digit.length !== 1 || !/^[0-9]$/.test(digit)) {
+    return digit;
+  }
+  const subscriptDigits: { [key: string]: string } = {
+    '0': '₀',
+    '1': '₁',
+    '2': '₂',
+    '3': '₃',
+    '4': '₄',
+    '5': '₅',
+    '6': '₆',
+    '7': '₇',
+    '8': '₈',
+    '9': '₉',
+  };
+  return subscriptDigits[digit] || digit;
+};
+
+const toSuperscript = (str: string | number): string => {
+  'worklet';
+  const char = String(str);
+  const superscriptDigits: { [key: string]: string } = {
+    '0': '⁰',
+    '1': '¹',
+    '2': '²',
+    '3': '³',
+    '4': '⁴',
+    '5': '⁵',
+    '6': '⁶',
+    '7': '⁷',
+    '8': '⁸',
+    '9': '⁹',
+    '-': '⁻',
+  };
+  return superscriptDigits[char] || char;
+};
```

Comment:
Should move the object definitions outside of the function. Nit but should prefer named function where parent context is not required.